libcalico-go/lib/clientv3/tier.go

```diff
@@ -131,7 +146,10 @@ func (r tiers) Delete(ctx context.Context, name string, opts options.DeleteOptio
 func (r tiers) Get(ctx context.Context, name string, opts options.GetOptions) (*apiv3.Tier, error) {
 	out, err := r.client.resources.Get(ctx, opts, apiv3.KindTier, noNamespace, name)
 	if out != nil {
-		return out.(*apiv3.Tier), err
+		res := out.(*apiv3.Tier)
+		// Default values when reading from backend.
+		cresources.DefaultTierFields(res)
```

Comment:
I don't think we need to do this defaulting here if the backend code has already done it